client/client.go

```diff
@@ -37,22 +37,32 @@ type LogClient struct {
 	LogID    int64
 	client   trillian.TrillianLogClient
 	hasher   merkle.TreeHasher
-	STR      trillian.SignedLogRoot
-	MaxTries int
+	str      trillian.SignedLogRoot
+	maxTries int
 	pubKey   gocrypto.PublicKey
 }
 
 // New returns a new LogClient.
-func New(logID int64, client trillian.TrillianLogClient, hasher merkle.TreeHasher, pubKey gocrypto.PublicKey) *LogClient {
+func New(logID int64, client trillian.TrillianLogClient, hasher merkle.TreeHasher, pubKey gocrypto.PublicKey) LogVerifier {
 	return &LogClient{
 		LogID:    logID,
 		client:   client,
 		hasher:   hasher,
-		MaxTries: 3,
+		maxTries: 3,
 		pubKey:   pubKey,
 	}
 }
 
+// STR returns the current trusted SignedLogRoot
```

Comment:
'trusted' is slightly incorrect as there is no consistency proof check from the first root read from the log.

If the log has a size > 1 you could ask for a consistency proof from 1 to current size and verify that it's correct rather than always skipping for the first root. I can't comment on that bit of code thanks to Github.